File: 10_02_2024_rotate_and_delete.py

  
from collections import deque

class Solution:
    def rotateDelete(self, arr):
        i = 1
        n = len(arr)
        arr = deque(arr)
        
        while i <= n // 2:
            arr.rotate(1)
            del arr[-i]
            i += 1
        
        return arr[0]

   
        

File: 10_02_2024_rotate_and_delete.java


class Solution {
    public int rotateDelete(List<Integer> arr) {
        int i = 1;
        int n = arr.size();
        while (i < (n / 2) + 1) {
            Collections.rotate(arr, 1);
            arr.remove(arr.size() - i);
            i++;
        }
        return arr.get(0);
    }

}
